src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
The `getAll` method should be parameterized to return `List<T>` instead of `List<Machine>`. This will ensure that the method returns a list of the specific type `T` that implements this interface, providing better type safety and aligning with the generic type parameter `T`.